core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/security/KeyStoreMetadata.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.core.security;
+
+/**
+ * This class holds the metadata of a keystore.
+ */
+public class KeyStoreMetadata {
+
+    private String keyStoreName;
+    private String keyStoreType;
+    private String provider;
+    private String pubKeyFilePath;
```

Comment:
Better to avoid doing this file system drama at the kernel level.. If there are no perf implications, can we just pass the public key and keep this model object clean.. (without leaking the implementation details like keeping temp files to the model objects)

Reply:
Addressed with https://github.com/wso2/carbon-kernel/pull/4099/commits/f89bbb95d862ca9f626a21592685bbd3e230eb11